bindings.c

```diff
@@ -4034,20 +4067,47 @@ static int read_cpuacct_usage_all(char *cg, char *cpuset, struct cpuacct_usage *
 	ticks_per_sec = sysconf(_SC_CLK_TCK);
 
 	if (ticks_per_sec < 0 && errno == EINVAL) {
-		lxcfs_debug(
+		lxcfs_v(
 			"%s\n",
 			"read_cpuacct_usage_all failed to determine number of clock ticks "
 			"in a second");
 		return -1;
 	}
 
 	cpu_usage = malloc(sizeof(struct cpuacct_usage) * cpucount);
+	bzero(cpu_usage,sizeof(struct cpuacct_usage) * cpucount);
 	if (!cpu_usage)
 		return -ENOMEM;
 
 	if (!cgfs_get_value("cpuacct", cg, "cpuacct.usage_all", &usage_str)) {
-		rv = -1;
-		goto err;
+		lxcfs_v("failed to read cpuacct.usage_all. usage_str: %s\n", usage_str);
+
+		// read cpuacct.usage_percpu instead
+		lxcfs_v("reading cpuacct.usage_percpu instead%s\n", "");
+		if (!cgfs_get_value("cpuacct", cg, "cpuacct.usage_percpu", &usage_str)) {
+			rv = -1;
+			goto err;
+		}
+
+		// convert cpuacct.usage_percpu into cpuacct.usage_all
+		lxcfs_v("converting cpuacct.usage_percpu into cpuacct.usage_all%s\n", "");
+		StringBuilder *sb = sb_create();
+		sb_append(sb, "cpu user system\n");
+		int i = 0;
+		while(sscanf(usage_str + read_pos, "%lu %n", &cg_user, &read_cnt) > 0){
```

Comment:
Missing space between `while` keyword and opening `(`as well as missing space after `)` and `{`